research/api-comparision.md

```diff
@@ -0,0 +1,619 @@
+*/
+
+/**
+* LaunchDarkly
+*
+* SDK Repo: https://github.com/launchdarkly/java-server-sdk
+*/
+int intVariation(String featureKey, LDUser user, int defaultValue)
+double doubleVariation(String featureKey, LDUser user, double defaultValue)
+// Response also contains evaluation details
+EvaluationDetail<Integer> intVariationDetail(String featureKey, LDUser user, int defaultValue)
+EvaluationDetail<Double> doubleVariationDetail(String featureKey, LDUser user, double defaultValue)
+
+/**
+* Split
+*
+* SDK Repo: https://github.com/splitio/java-client
+*
+* NOTE: Not supported; strings can be parsed into numeric values
+*/
+
+/**
+* CloudBees Rollout
+*
+* SDK Repo: N/A
+*/
+public class Container implements RoxContainer {
+    public final RoxInt titleSize = new RoxInt(5, new int[]{ 8, 13 });
+    public final RoxDouble specialNumber = new RoxDouble(3.14, new double[]{ 2.71, 0.577 });
+}
+
+Container flags = new Container();
+Rox.register("test-namespace", flags);
+flags.titleColors.getValue();
+
+// Dynamic API
+Rox.dynamicApi.getNumber('ui.textSize', 12);
+Rox.dynamicApi.getNumber('ui.textColor', 18, context);
+
+/**
+* Harness
+*
+* SDK Repo: https://github.com/harness/ff-java-server-sdk
+*/
+double numberVariation(String key, Target target, int defaultValue)
+```
+
+</details>
+
+<details>
+  <summary>NodeJS</summary>
+
+```typescript
+/**
+* Unleash
+*
+* SDK Repo: https://github.com/Unleash/unleash-client-node
+*
+* NOTE: Not supported; Strings can be parsed into numeric values.
+*/
+
+/**
+* Flagsmith
+*
+* SDK Repo: https://github.com/Flagsmith/flagsmith-nodejs-client
+*/
+getValue(key: string): Promise<string | number | boolean>;
+getValue(key: string, userId: string): Promise<string | number | boolean>;
+
+/**
+* LaunchDarkly
+*
+* SDK Repo: https://github.com/launchdarkly/node-client-sdk
+*
+* Note: TS typings assign LDFlagValue = any;
+* Note: variation values can be defined as boolean, number, or string. Documentation suggests casting: https://docs.launchdarkly.com/sdk/client-side/node-js#getting-started
+*/
+variation(
+key: string,
+  user: LDUser,
+  defaultValue: LDFlagValue,
+  callback?: (err: any, res: LDFlagValue) => void
+): Promise<LDFlagValue>
+// Response also contains evaluation details
+variationDetail(
+  key: string,
+  user: LDUser,
+  defaultValue: LDFlagValue,
+  callback?: (err: any, res: LDEvaluationDetail) => void
+): Promise<LDEvaluationDetail>;
+
+/**
+* Split
+*
+* SDK Repo: https://github.com/splitio/javascript-client
+*
+* NOTE: Not supported; Strings can be parsed into numeric values.
+* NOTE: TS typings assign Treatment = string;
+*/
+
+/**
+* CloudBees Rollout
+*
+* SDK Repo: N/A
+*
+* NOTE: Flags are configured as code and contain default values.
+*
+*/
+const flags = {
+  titleSize: new RoxNumber(12, [12, 14, 18, 24])
+};
+
+Rox.register('test-namespace', flags);
+flags.titleSize.value();
+
+export class RoxNumber {
+  getValue(context?: unknown): number;
+}
+
```

Comment:
I couldn't find any reference to dynamic APIs in Node. Perhaps it's worth confirming and adding a note here since it's available in Java.